File: Server/Server.NetWork/Messages/MessageObject.cs
using Newtonsoft.Json;
using ProtoBuf;

namespace Server.NetWork.Messages
{
    [ProtoContract]
    public abstract class MessageObject : IMessage
    {
        /// <summary>
        /// 消息唯一id
        /// </summary>
        [JsonIgnore]
        public int UniId { get; set; }

        /// <summary>
        /// 消息ID
        /// </summary>
        [ProtoMember(998)]
        public int MessageId { get; set; }

        public override string ToString()
        {
            return JsonConvert.SerializeObject(this);
        }

        /// <summary>
        /// 消息的唯一ID
        /// </summary>
        [ProtoMember(999)]
        public long UniqueId { get; set; }

        public MessageObject()
        {
            UniqueId = Utility.UtilityIdGenerator.GetNextUniqueId();
        }
    }
}

File: LayaBox/src/gameframex/network/MessageObject.ts
import UtilityIdGenerator from "../utility/UtilityIdGenerator";
import IMessage from "./IMessage";

export default class MessageObject {
    /**
     * 唯一标识,请不要外部修改
     */
    public UniqueId: number = 0;

    /**
     * 消息ID.请不要外部修改.
     */
    public MessageId: number = 0;

    /**
     *
     */
    constructor() {
        this.UniqueId = UtilityIdGenerator.getUniqueIntId();
    }
}
